openapi_python_client/templates/property_templates/union_property.py.jinja

```diff
@@ -1,3 +1,36 @@
+{% macro construct_inner_property(inner_property) %}
+{% import "property_templates/" + inner_property.template as inner_template %}
+{% if inner_template.check_type_for_construct %}
+if not {{ inner_template.check_type_for_construct(inner_property, "data") }}:
+    raise TypeError()
+{% endif %}
+{{ inner_template.construct(inner_property, "data") }}
+return {{ inner_property.python_name }}
+{%- endmacro %}
+
+{% macro construct_discriminator_lookup(property) %}
+{% set _discriminator_properties = [] -%}
+{% for discriminator in property.discriminators %}
```

Comment:
As a result of the new logic in `union.py`, `property.discriminators will look like this:
```python
[
  {
    "property_name": <name of the discriminator property>,
    "value_to_model_map": {
      "somePossibleValue": <ModelProperty instance corresponding to that value>
    }
  }
]
```

There can be multiple discriminators due to the nested-unions situation I described in `union.py`.